core/src/scene/sceneLoader.cpp

```diff
@@ -73,21 +73,15 @@ void SceneLoader::loadShaderConfig(Node shaders, Style& style, Scene& scene) {
     if (!shaders) { return; }
 
     if (Node extensionsNode = shaders["extensions"]) {
-        std::vector<std::string> extensions;
-        if (extensionsNode.IsScalar()) {
-            extensions.push_back(extensionsNode.as<std::string>());
-        } else if(extensionsNode.IsSequence()) {
-            extensions.reserve(extensionsNode.size());
+        if(extensionsNode.IsSequence()) {
             for (const auto& extNode : extensionsNode) {
-                extensions.push_back(extNode.as<std::string>());
+                char buffer[1000]; //sufficient space
```

Comment:
We should consider using https://github.com/c42f/tinyformat as typesafe c++11 printf replacement. It is likely to be more efficient than our common string concatenation and much nicer to write formatting specifiers with it.

Just tests: When adding a version using std::string concatenation it is actually slower, even without precision formatting specifiers
https://github.com/c42f/tinyformat/blob/master/tinyformat_speed_test.cpp